packages/api/src/auth.js

```diff
@@ -36,17 +36,34 @@ class Auth {
     return querystring.parse(respText);
   }
 
-  async retrieveUserData (accessToken) {
-    const result = await this.doFetch('https://api.github.com/user', {
+  async retrieveUserLogin (accessToken) {
+    const userData = await this.doFetch('https://api.github.com/user', {
       method: 'get',
-      headers: { 'content-type': 'application/json', 'User-Agent': 'flaky.dev', Authorization: 'token ' + accessToken }
+      headers: { 'content-type': 'application/json', 'User-Agent': 'flaky.dev', Authorization: `token ${accessToken}` }
     });
 
-    if (result.status !== 200) {
-      return null;
-    }
-    const resJSON = await result.json();
-    return resJSON;
+    if (userData.status !== 200) return null;
+
+    const data = await userData.json();
+    return data.login;
+  }
+
+  async retrieveUserPermission (accessToken, fullName) {
+    const login = await this.retrieveUserLogin(accessToken);
+
+    const result = await this.doFetch(`https://api.github.com/repos/${fullName}/collaborators/${login}/permission`, {
```

Comment:
awesome  👍 at the end of the day, I really like the route you've taken of calling APIs directly.

Reply:
Thanks! I did look a bit into Github's octokit for interacting with their API, but for our needs it just seemed simplest to stick with direct API calls.